src/librbd/image/RemoveRequest.cc

```diff
@@ -575,6 +577,11 @@ Context *RemoveRequest<I>::handle_mirror_image_remove(int *result) {
     *result = 0;
   }
 
+  if (m_from_trash_purge) {
+    // image was already removed from the directory
```

Comment:
Note: should remove image from directory and remove the "rbd_id.XYZ" object (so that I can move image XYZ to the trash and immediately create a new image named XYZ).